internal/controller/monovertexrollout/monovertexrollout_controller_test.go

```diff
@@ -256,6 +329,74 @@ func Test_processExistingMonoVertex_Progressive(t *testing.T) {
 				ctlrcommon.DefaultTestMonoVertexRolloutName + "-1": common.LabelValueUpgradeInProgress,
 			},
 		},
+		{
+			name:                    "AnalysisRun successful",
+			newControllerInstanceID: "1",
+			existingOriginalMonoVertexDef: *createMonoVertex(
+				numaflowv1.MonoVertexPhaseRunning,
+				numaflowv1.Status{},
+				map[string]string{
+					common.LabelKeyUpgradeState:  string(common.LabelValueUpgradePromoted),
+					common.LabelKeyParentRollout: ctlrcommon.DefaultTestMonoVertexRolloutName,
+				},
+				map[string]string{
+					common.AnnotationKeyNumaflowInstanceID: "0",
+				}),
+			existingUpgradeMonoVertexDef: ctlrcommon.CreateTestMonoVertexOfSpec(
+				monoVertexSpec, ctlrcommon.DefaultTestMonoVertexRolloutName+"-1",
+				numaflowv1.MonoVertexPhaseRunning,
+				numaflowv1.Status{
+					Conditions: []metav1.Condition{
+						{
+							Type:               string(numaflowv1.MonoVertexConditionDaemonHealthy),
+							Status:             metav1.ConditionTrue,
+							Reason:             "healthy",
+							LastTransitionTime: metav1.NewTime(time.Now()),
+						},
+					},
+				},
+				map[string]string{
+					common.LabelKeyUpgradeState:  string(common.LabelValueUpgradeInProgress),
+					common.LabelKeyParentRollout: ctlrcommon.DefaultTestMonoVertexRolloutName,
+				},
+				map[string]string{
+					common.AnnotationKeyNumaflowInstanceID: "1",
+				}),
+			initialRolloutPhase:       apiv1.PhasePending,
+			initialRolloutNameCount:   2,
+			initialInProgressStrategy: &progressiveUpgradeStrategy,
+			initialUpgradingChildStatus: &apiv1.UpgradingMonoVertexStatus{
+				UpgradingPipelineTypeStatus: apiv1.UpgradingPipelineTypeStatus{
+					UpgradingChildStatus: apiv1.UpgradingChildStatus{
+						Name:                ctlrcommon.DefaultTestMonoVertexRolloutName + "-1",
+						AssessmentStartTime: &metav1.Time{Time: time.Now().Add(-1 * time.Minute)},
+						AssessmentEndTime:   &metav1.Time{Time: time.Now().Add(-30 * time.Second)},
+						AssessmentResult:    apiv1.AssessmentResultSuccess,
```

Comment:
if you set this one to Unknown, does it take the MonoVertexRollout out of pending/in-progress-strategy?

Reply:
Yes, can make that change as well.